### DISCUSSION TITLE
Lines with length and angle?

### DISCUSSION THREAD

**Discussion by 68pandreas** [xournalpp/xournalpp#2505] [Q&A]
Hi, for drawings (e.g. vectors...) needed in my lectures it would be fine to be able to draw lines of certain angles and lengths, such as on a classical blackboard by use of a ruler. The simpliest would be if the line draw tool would tell me length in (prefereably) mm and angle in degree. Is this existing yet as somehow hidden feature, since I did not find it? Best regards, Andreas 

PS: Another maybe stupid question: vertical scrolling, is it just possible with a srollbar or should it also work with my fingers? This I could not yet manage to do so far though I was able to enable Pen/Stylus support on my Microsoft Surfage Go2.

**Comment by LittleHuba** [comment]
The ruler is an old question we would like to work on in the future. It will require some background work though and will not be in the near future. There was a proof of concept some while ago but I'm unable to find the ticket on short notice.

For support of scrolling using your fingers see #2435

**Comment by 68pandreas** [comment]
Thanks. So far I understand #2435 panning by fingers may come in the next version. 

Length and angle of a line: Most likely when drawing lines start end end point are known. So it should be straight forward to calculate length and angle and to show the values in the bottom line of the window, the length prefereable scaled. Best regards, Andreas

**Reply by LittleHuba** [reply]
The proposed ruler went a step further that you actually had a movable graphical ruler on the screen Where the pen would stick to if in the vincinity. But your request would of course be way easier.

@Febbe this sounds like another feature for the plugin API.
